### PR TITLE
TCM-434 | Connect home screen videos

### PR DESCRIPTION
**Pull request by devender197**
### Summary

- [x] Test button removed from home and displayed in settings.
- [x] Display cached user name.
- [x] Design specification improved.

### Issue

[TCM-434](https://hyfnla.atlassian.net/browse/TCM-434)

### Video/Screenshot

https://user-images.githubusercontent.com/24737954/120085913-0f097100-c091-11eb-96ad-ce6722b7d1b1.mov

### CODE REVIEW

File: app/src/main/java/com/trx/consumer/screens/home/HomeFragment.kt [hunk 1]
```diff
@@ -37,27 +38,26 @@ class HomeFragment : BaseFragment(R.layout.fragment_home) {
...
             eventLoadUser.observe(viewLifecycleOwner, handleLoadUser)
             eventLoadBanner.observe(viewLifecycleOwner, handleLoadBanner)
 
-            eventTapBanner.observe(viewLifecycleOwner, handleTapBanner)
-            eventTapUser.observe(viewLifecycleOwner, handleTapUser)
             eventShowVideo.observe(viewLifecycleOwner, handleShowVideo)
-
             eventShowPromo.observe(viewLifecycleOwner, handleShowPromo)
             eventShowHud.observe(viewLifecycleOwner, handleShowHud)
+            eventShowEditProfile.observe(viewLifecycleOwner, handleTapEditProfile)
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/home/HomeFragment.kt:58]: These should be `eventShowProfile` and `handleShowProfile` I think, EditProfile is our `Update` screen

File: app/src/main/res/layout/fragment_home.xml [hunk 2]
```diff
@@ -53,6 +39,18 @@
                     app:layout_constraintTop_toTopOf="parent"
                     tools:text="Welcome, Kristoph" />
 
+                <com.trx.consumer.common.CommonLabel
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_home.xml:42]: Don't think this is in v0: [figma](https://www.figma.com/file/LXa0oBt9aSNOk0YMAeSM8Q/TRX-Consumer?node-id=3473%3A573)

File: app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt [hunk 3]
```diff
@@ -98,15 +94,22 @@ class HomeViewModel @ViewModelInject constructor(
...
+            if (response.isSuccess) {
+                val model = VideosResponseModel.parse(response.responseString)
+                eventLoadVideos.postValue(model.workouts)
+            }
+        }
     }
 
-    fun doTapBanner() {
-        eventTapBanner.postValue(bannerUrlString)
+    fun doTapEditProfile() {
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt:107]: Related to [this](https://github.com/hyfn/trx-consumer-android/pull/87/files#r641983168)

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Couple small things

File: app/src/main/java/com/trx/consumer/screens/home/HomeFragment.kt (hunk 1 above)
Comment (devender197) [reply app/src/main/java/com/trx/consumer/screens/home/HomeFragment.kt:58]: Thank you @VivekV95 . I did this to make parity with iOS. @onodude if we implement this suggestion we need to make changes at iOS side also.

Comment (onodude) [reply app/src/main/java/com/trx/consumer/screens/home/HomeFragment.kt:58]: we can remove these events

File: app/src/main/res/layout/fragment_home.xml (hunk 2 above)
Comment (devender197) [reply app/src/main/res/layout/fragment_home.xml:42]: It is present in [iOS here](https://github.com/hyfn/trx-consumer-ios/blob/develop/TRX-Consumer/Screens/Home/HomeView.swift#L151). As we don't follow figma all the time. Hence to make parity I have created it. @onodude Please let me know we need to remove it or hide it.

File: app/src/main/java/com/trx/consumer/screens/home/HomeFragment.kt (hunk 1 above)
Comment (devender197) [reply app/src/main/java/com/trx/consumer/screens/home/HomeFragment.kt]: Changed the `handleTapEditProfile` to `handleShowEditProfile` [here](https://github.com/hyfn/trx-consumer-android/pull/87/commits/b9d125410067c30cc23eabd6c63dd4788e5564fc)

File: app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt (hunk 3 above)
Comment (devender197) [reply app/src/main/java/com/trx/consumer/screens/home/HomeViewModel.kt]: [Done](https://github.com/hyfn/trx-consumer-android/pull/87/commits/df67225a103a84aca807715529b4cde906bc162d)

File: app/src/main/res/layout/fragment_home.xml (hunk 2 above)
Comment (devender197) [reply app/src/main/res/layout/fragment_home.xml]: Removed

**Review by onodude — APPROVED** [review]
looks good @devender197 thanks!

**Pull request merged**